src/tiny_slam/tiny_world.h

```diff
@@ -13,30 +13,49 @@
 #include "tiny_grid_cells.h"
 #include "tiny_scan_matcher.h"
 
+/*!
+ * \brief Structure-storage of two different quality of laser scanner - two values of trust for laser data
+ */
 struct TinyWorldParams {
   double localized_scan_quality, raw_scan_quality;
 };
 
+/*!
+ * \brief Derived class from LaserScanGridWorld to estimate occupancy of each cell
+ *
+ * There is an robot state handle based on used scanner matcher rules and laser data handle based on algorithm from the article with wall blur
+ */
 class TinyWorld : public LaserScanGridWorld {
 private: // internal params
   // Scan matcher
-  const double SIG_XY = 0.2;
-  const double SIG_TH = 0.1;
-  const double BAD_LMT = 20;
-  const double TOT_LMT = BAD_LMT * 5;
+  const double SIG_XY = 0.2; ///< data member \f$\sigma_{x,y}\f$
+  const double SIG_TH = 0.1; ///< data member \f$\sigma_{\theta}\f$
+  const double BAD_LMT = 20; ///< amount of bad steps for monte-carlo choice better robot position
+  const double TOT_LMT = BAD_LMT * 5; ///< maximum amount of steps are able to be done
 
-  const double HOLE_WIDTH = 1.5;
+  const double HOLE_WIDTH = 1.5; ///< the total width
 public:
   using Point = DiscretePoint2D;
 public:
 
+  /*!
+   * Parameterized constructor sets all data members
+   * \param[in] gcs    - shared pointer on strategy for each cell (how does estimate occupancy, cost of laser data etc)
+   * \param[in] params - the initial values of trust for laser data
+   */
   TinyWorld(std::shared_ptr<GridCellStrategy> gcs,
             const TinyWorldParams &params) :
     LaserScanGridWorld(gcs), _gcs(gcs), _params(params),
     _scan_matcher(new TinyScanMatcher(_gcs->cost_est(),
                                       BAD_LMT, TOT_LMT,
                                       SIG_XY, SIG_TH)) {}
 
+  /*!
+   * Updates robot pose and map
```

Comment:
Consider: "<...> by prediction-correction scheme.

The scan quality used for map update depends on whether the robot pose have been changed during correction step."